lib/rack/urlmap.rb

```diff
@@ -20,9 +20,11 @@ def initialize(map = {})
     end
 
     def remap(map)
+      @known_hosts = []
```

Comment:
Should we make this a `Set`?  I'm not sure how many known hosts a typical app is going to deal with, so maybe an array is fine.

Reply:
A `Set` isn't a bad shout at all.